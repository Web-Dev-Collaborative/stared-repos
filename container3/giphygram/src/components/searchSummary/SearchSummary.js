import React from 'react';
import PropTypes from 'prop-types';

export class SearchSummary extends React.Component {
  static propTypes = {
    total: PropTypes.number,
  };

  static defaultProps = {
    total: null,
  };

  render() {
    const { total } = this.props;

    if (total === null) {
      return null;
    }

    return (
      <div className="mb-3">
        <small>
          Total results:
          &nbsp;
          <span className="badge badge-dark">
            {total}
          </span>
        </small>
      </div>
    );
  }
}
