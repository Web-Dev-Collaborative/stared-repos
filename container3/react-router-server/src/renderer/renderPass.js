import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
