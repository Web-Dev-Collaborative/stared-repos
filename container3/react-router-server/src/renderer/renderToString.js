import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
