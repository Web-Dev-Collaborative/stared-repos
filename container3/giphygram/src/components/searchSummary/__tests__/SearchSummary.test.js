import React from 'react';
import renderer from 'react-test-renderer';
import { SearchSummary } from '../SearchSummary';

describe('SearchSummary', () => {
  it('should be rendered correctly', () => {
    const tree = renderer
      .create(<SearchSummary total={42} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should not be rendered if there are no search items', () => {
    const tree = renderer
      .create(<SearchSummary />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered if nothing was found', () => {
    const tree = renderer
      .create(<SearchSummary total={0} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
