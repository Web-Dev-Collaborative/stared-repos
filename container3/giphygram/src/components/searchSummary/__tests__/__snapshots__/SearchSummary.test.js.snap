// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSummary should be rendered correctly 1`] = `
<div
  className="mb-3"
>
  <small>
    Total results:  
    <span
      className="badge badge-dark"
    >
      42
    </span>
  </small>
</div>
`;

exports[`SearchSummary should be rendered if nothing was found 1`] = `
<div
  className="mb-3"
>
  <small>
    Total results:  
    <span
      className="badge badge-dark"
    >
      0
    </span>
  </small>
</div>
`;

exports[`SearchSummary should not be rendered if there are no search items 1`] = `null`;
